packages/odf/components/s3-browser/create-bucket/CreateBucketForm.tsx

```diff
@@ -0,0 +1,171 @@
+    try {
+      const tagSet: PutBucketTaggingCommandInput['Tagging']['TagSet'] = tagsData
+        .filter((pair: string[]) => !_.isEmpty(pair[0]))
+        .map((pair: string[]) => ({ Key: pair[0], Value: pair[1] }));
+      if (!_.isEmpty(tagSet)) {
+        await noobaaS3.putBucketTags({
+          Bucket: bucketName,
+          Tagging: { TagSet: tagSet },
+        });
+      }
+    } catch ({ name, message }) {
+      // @TODO: add a toast warning that is visible after redirection to bucket details.
+      // eslint-disable-next-line no-console
+      console.error(`Error while updating bucket tags: ${name}: ${message}`);
+    }
+    navigate(`${BUCKETS_BASE_ROUTE}/${bucketName}`);
+  };
+
+  return (
+    <Form onSubmit={handleSubmit(save)} className="pf-v5-u-w-50">
+      <TextInputWithFieldRequirements
+        control={control}
+        fieldRequirements={fieldRequirements}
+        popoverProps={{
+          headerContent: t('Name requirements'),
+          footerContent: `${t('Example')}: my-bucket`,
+        }}
+        formGroupProps={{
+          label: t('Bucket Name'),
+          fieldId: 'bucket-name',
+          isRequired: true,
+          className: 'control-label',
+        }}
+        textInputProps={{
+          id: 'bucket-name',
+          name: 'bucketName',
+          className: 'pf-v5-c-form-control',
+          type: 'text',
+          placeholder: t('my-bucket'),
+          'aria-describedby': 'bucket-name-help',
+          'data-test': 'bucket-name',
+        }}
+        helperText={t('A unique name for your bucket.')}
+      />
+      <FormGroup
+        label={t('Tags')}
+        labelInfo={t('Use different criteria for tagging your bucket.')}
+        className={cn('odf-create-s3-bucket-form__tags', {
+          'odf-create-s3-bucket-form__tags--empty': _.isEmpty(tagsData),
+        })}
+      >
+        {_.isEmpty(tagsData) && (
+          <div className="pf-v5-u-disabled-color-100 pf-v5-u-mr-sm">
+            {t('No tags are attached to this bucket.')}
+          </div>
+        )}
+        <LazyNameValueEditor
+          className="pf-v5-u-font-weight-bold pf-v5-u-font-size-sm"
+          addString={t('Add tag')}
+          valueString={t('Value (optional)')}
+          nameValuePairs={tagsData}
+          updateParentData={({ nameValuePairs }) => {
+            setTagsData(nameValuePairs);
+          }}
+          hideHeaderWhenNoItems={true}
+          IconComponent={TagIcon}
+          nameMaxLength={128}
+          valueMaxLength={256}
+        />
+      </FormGroup>
+      {!isValid && isSubmitted && (
+        <Alert
+          variant="danger"
+          isInline
+          title={t('Address form errors to proceed')}
+        />
+      )}
+      <ButtonBar errorMessage={errorMessage} inProgress={inProgress}>
+        <ActionGroup className="pf-v5-c-form">
+          <Button
+            id="create-s3-bucket-btn"
+            type="submit"
+            variant="primary"
```

Comment:
can be addressed in followup PR...
```suggestion
            variant={ButtonVariant.primary}
```